src/Interpreters/ThreadStatusExt.cpp

```diff
@@ -406,6 +406,11 @@ void ThreadStatus::detachQuery(bool exit_if_already_detached, bool thread_exits)
     query_context.reset();
     thread_trace_context.trace_id = 0;
     thread_trace_context.span_id = 0;
+
+    /// Avoid leaking of ThreadGroupStatus::finished_threads_counters_memory
+    /// (this is in case someone uses system thread but did not call getProfileEventsCountersAndMemoryForThreads())
+    thread_group->getProfileEventsCountersAndMemoryForThreads();
```

Comment:
Need to rename `getProfileEventsCountersAndMemoryForThreads` in something like `getProfileEventsCountersAndMemoryForThreads`, because it is unexpected that get method will cleanup something.

Reply:
Egh, seems that you made a typo here, I will rename it to `resetProfileEventsCountersAndMemoryForThreads`

P.S. Usually I'm trying to keep bug fixes patches as small as possible to avoid possible conflicts during backporting.